ballerina/validator_visitor.bal

```diff
@@ -227,12 +294,78 @@ class ValidatorVisitor {
             }
         } else {
             string expectedTypeName = getOfType(inputValue.'type).name.toString();
-            string message = string`${expectedTypeName} cannot represent non ${expectedTypeName} value: {}`;
+            string listError = getListElementError(self.argumentPath);
+            string message = string`${listError}${expectedTypeName} cannot represent non ${expectedTypeName} value: {}`;
+            ErrorDetail errorDetail = getErrorDetailRecord(message, location);
+            self.errors.push(errorDetail);
+        }
+    }
+
+    isolated function validateListVariableValue(json[] variableValues, __InputValue inputValue,
+                                                Location location, string fieldName) {
+        if getTypeKind(inputValue.'type) == LIST {
+            __InputValue listItemInputValue = createInputValueForListItem(inputValue);
+            if variableValues.length() > 0 {
+                foreach int i in 0..< variableValues.length() {
+                    self.updatePath(i);
+                    json listItemValue = variableValues[i];
+                    if listItemValue is Scalar {
+                        parser:ArgumentValue argValue = listItemValue;
+                        if getOfType(listItemInputValue.'type).kind == ENUM {
+                            self.validateEnumArgument(argValue, location, ENUM, listItemInputValue);
+                        } else {
+                            string expectedTypeName = getOfType(listItemInputValue.'type).name.toString();
+                            if expectedTypeName == FLOAT && argValue is int {
```

Comment:
Isn't the `coerceArgumentIntValueToFloat` function does the same (or something similar)?

Reply:
`coerceArgumentIntValueToFloat` function is used for argument nodes. Since this is a variable value, here we get a `json[]`. Shall we have a separate function? This is the only place we get a `json[]` member to coerce.

Reply:
Right. In that case, this will be okay.